sdk/eventhub/event-hubs/src/util/connectionStringUtils.ts

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { parseConnectionString } from "@azure/core-amqp";
+
+/**
+ * The set of properties that comprise an Event Hub connection string.
+ */
+export interface EventHubConnectionStringProperties {
+  /**
+   * The fully qualified Event Hub namespace extracted from the "Endpoint" in the
+   * connection string. This is likely to be similar to "{yournamespace}.servicebus.windows.net".
+   * This is typically used to construct the EventHub{Producer|Consumer}Client.
+   */
+  fullyQualifiedNamespace: string;
+  /**
+   * The value for "Endpoint" in the connection string.
+   */
+  endpoint: string;
+  /**
+   * The value for "EntityPath" in the connection string.
+   * Connection string from a Shared Access Policy created at the namespace level
+   * will not have the EntityPath in it.
+   */
+  eventHubName?: string;
+  /**
+   * The value for "SharedAccessKey" in the connection string. This along with the "SharedAccessKeyName"
+   * in the connection string is used to generate a SharedAccessSignature which can be used authorize
+   * the connection to the service.
+   */
+  sharedAccessKey?: string;
+  /**
+   * The value for "SharedAccessKeyName" in the connection string. This along with the "SharedAccessKey"
+   * in the connection string is used to generate a SharedAccessSignature which can be used authorize
+   * the connection to the service.
+   */
+  sharedAccessKeyName?: string;
+  /**
+   * The value for "SharedAccessSignature" in the connection string. This is typically not present in the
+   * connection string generated for a Shared Access Policy. It is instead generated by the
+   * user and appended to the connection string for ease of use.
+   */
+  sharedAccessSignature?: string;
+}
+
+/**
+ * Parses given connection string into the different properties applicable to Azure Event Hubs.
+ * The properties are useful to then construct an EventHub{Producer|Consumer}Client.
+ * @param connectionString The connection string associated with the Shared Access Policy created
+ * for the Event Hubs namespace.
+ */
+export function parseEventHubConnectionString(
+  connectionString: string
+): Readonly<EventHubConnectionStringProperties> {
+  const parsedResult = parseConnectionString<GenericConnectionStringProperties>(connectionString);
+
+  const validationResult = validate(parsedResult);
+
+  if (!validationResult.isValid) {
+    throw new Error(validationResult.message);
+  }
+
+  const output: EventHubConnectionStringProperties = {
+    fullyQualifiedNamespace: (parsedResult.Endpoint.match(".*://([^/]*)") || [])[1],
```

Comment:
I was thinking to use `new URL(parsedResult.Endpoint).hostname` but wanted to keep the same behavior as servicebus. In servicebus parser, if the endpoint doesn't match this regex `fullyQualifiedNamespace` will end up `undefined` whereas `new URL` constructor will raise an error at this point if the Endpoint isn't a valid URL. Wasn't sure if folks have a preference so I matched existing behavior here but let me know if you feel I should change this